processor/metricstransformprocessor/metrics_transform_processor_test.go

```diff
@@ -239,54 +65,92 @@ func TestMetricsTransformProcessor(t *testing.T) {
 			require.Equal(t, 1, len(got))
 			gotMD := pdatautil.MetricsToMetricsData(got[0])
 			require.Equal(t, 1, len(gotMD))
-			require.Equal(t, len(test.outMN), len(gotMD[0].Metrics))
+			actualOut := gotMD[0].Metrics
+			require.Equal(t, len(test.out), len(actualOut))
 
-			targetNameToTransform := make(map[string]Transform, len(test.transforms))
-			for _, transform := range test.transforms {
-				targetName := transform.MetricName
-				if transform.NewName != "" {
-					targetName = transform.NewName
-				}
-				targetNameToTransform[targetName] = transform
-			}
-			for idx, out := range gotMD[0].Metrics {
-				// check name
-				assert.Equal(t, test.outMN[idx], out.MetricDescriptor.Name)
-				// check labels
-				// check the updated or inserted is correctly updated
-				transform, ok := targetNameToTransform[out.MetricDescriptor.Name]
-				if ok {
-					for lidx, l := range out.MetricDescriptor.LabelKeys {
-						assert.Equal(t, test.outLabels[lidx], l.Key)
-					}
-				}
-				// check the original is untouched if insert
-				if transform.Action == Insert && out.MetricDescriptor.Name == transform.MetricName {
-					for lidx, l := range out.MetricDescriptor.LabelKeys {
-						assert.Equal(t, test.inLabels[lidx], l.Key)
-					}
-				}
+			for idx, out := range test.out {
+				assert.Equal(t, *out, *actualOut[idx])
 			}
+
 			assert.NoError(t, mtp.Shutdown(ctx))
 		})
 	}
 }
 
+func TestComputeDistVals(t *testing.T) {
+	t.Run("distribution value calculation test", func(t *testing.T) {
+		// next stores the results of the aggregation metric processor.
+		next := &etest.SinkMetricsExporter{}
+		cfg := &Config{
+			ProcessorSettings: configmodels.ProcessorSettings{
+				TypeVal: typeStr,
+				NameVal: typeStr,
+			},
+		}
+
+		mtp := newMetricsTransformProcessor(next, cfg, nil)
+		assert.NotNil(t, mtp)
+
+		caps := mtp.GetCapabilities()
+		assert.Equal(t, true, caps.MutatesConsumedData)
+		ctx := context.Background()
+		assert.NoError(t, mtp.Start(ctx, nil))
+
+		pointGroup1 := []float64{1, 2, 3, 7, 4}
+		pointGroup2 := []float64{1, 2, 3, 3, 1}
```

Comment:
I'd pull these out into input & output parameters so it's easy to configure a couple of test cases.

Side note: No need to use `t.Run` if you only have one test case.

Reply:
Updated this to remove `t.Run`, but I don't quite get the "pulling out into input & output parameters". Do you mean creating custom testcase struct for these points?

Reply:
Yes exactly.

Sorry that wasn't very clear - I meant I would recommend creating multiple test cases instead of removing `t.Run`, but was just commenting that `t.Run` is superfluous if you only have one test

Reply:
Got it. Updated